src/plone/restapi/services/aliases/get.py

```diff
@@ -0,0 +1,83 @@
+from plone.app.redirector.interfaces import IRedirectionStorage
+from plone.restapi.interfaces import IExpandableElement
+from plone.restapi.serializer.converters import datetimelike_to_iso
+from plone.restapi.services import Service
+from Products.CMFPlone.controlpanel.browser.redirects import RedirectsControlPanel
+from Products.CMFPlone.interfaces import IPloneSiteRoot
+from zope.component import adapter
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+
+
+@implementer(IExpandableElement)
+@adapter(Interface, Interface)
+class Aliases:
+    """@aliases"""
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    def reply_item(self):
+        storage = getUtility(IRedirectionStorage)
+        context_path = "/".join(self.context.getPhysicalPath())
+        redirects = storage.redirects(context_path)
+        aliases = [deroot_path(alias) for alias in redirects]
+        self.request.response.setStatus(201)
+        return [{"path": alias} for alias in aliases]
+
+    def reply_root(self):
+        """
+        redirect-to - target
+        path        - path
+        redirect    - full path with root
+        """
+        batch = RedirectsControlPanel(self.context, self.request).redirects()
+        redirects = [entry for entry in batch]
+
+        for redirect in redirects:
+            del redirect["redirect"]
+            redirect["datetime"] = datetimelike_to_iso(redirect["datetime"])
+        self.request.response.setStatus(201)
+
+        self.request.form["b_start"] = "0"
+        self.request.form["b_size"] = "1000000"
```

Comment:
I am wondering if we shouldn't go the extra mile of implementing batching here. I can imagine that the number of redirects can become quite significant even on mid-size websites.

Reply:
@tisto Do you suggest to set the default `b_size` to `50`?